### PR TITLE
CI: Add Pull Request format check

### PR DESCRIPTION
**Pull request by gxalpha**
### Description
Adds checks to make sure that the code is not on the master branch and that 
the commits are properly formatted.

I wanted to cleanly keep all jobs one workflow, while not having endless skipped runs, thus the workflow
ignores pushes to the master branch.
As a side effect, the clang format will no longer run on that branch. However, this change also has been proposed in PatTheMavs cmake overhaul and I seemed to not get rejected yet, so I'll assume thats fine. Please correct me if I'm wrong.

### Motivation and Context
An idea like this was discussed on the Discord *a long time ago*, but no one implemented it. Here you go.

### How Has This Been Tested?
Tested on my fork:

Wrongly formatted commits: https://github.com/gxalpha/obs-studio/pull/9
Correctly formatted commits: https://github.com/gxalpha/obs-studio/pull/10
~~PR from master branch: https://github.com/gxalpha/obs-studio/pull/12~~ Check got removed, see below

### Types of changes
- New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by WizardCM** [comment]
I personally think the format check should be separate from other checklist items, and then the rest should be one check (titled "PR Checklist" for example).

Additionally, I'm not 100% sure on making this dependent on being on the master branch purely based on our past experience - users who did use the `master` branch usually have no idea how to use Git, and ended up creating multiple new PRs trying to fix it.

**Comment by gxalpha** [comment]
> I personally think the format check should be separate from other checklist items, and then the rest should be one check (titled "PR Checklist" for example).

Alright
Also "checklist" is the word I was looking yesterday the entire evening, thanks :D

> Additionally, I'm not 100% sure on making this dependent on being on the master branch purely based on our past experience - users who did use the master branch usually have no idea how to use Git, and ended up creating multiple new PRs trying to fix it.

Would you suggest adding more information (maybe a link to a Git beginners guide) or just removing that check?
After all, if you want to change the head branch of the PR, there (to my knowledge) is no way other than creating a new PR (only one obviously, I agree it would be nice to prevent multiple)

**Comment by WizardCM** [comment]
A Git beginners guide, and one that provides a super basic guide on rebasing wouldn't be too bad - but in this case honestly removing the `master` branch check is probably fine for now because you are correct, it requires closing & reopening the PR.

**Comment by gxalpha** [comment]
I'll leave it commented out for now then

**Comment by PatTheMav** [comment]
As you mentioned I made the code format checks a mandatory step in the main workflow in my rework, that's where I'd put such a check as well (and also just use `curl` and `jq` to parse the JSON, no need to use yet another third party action whose source code usually nobody audits).

I'm not fully sold on the necessity of checking this via CI but not opposed to it if enough people feel it's worthwhile.

File: .github/workflows/pull-request-format.yml
Comment (RytoEX) [inline .github/workflows/pull-request-format.yml:36]: There are occasions where [the commit prefix contains a slash](https://github.com/obsproject/obs-studio/commit/ce6f233e2d026ea5373a74d245307492788346ad) (see also [one with a comma and slash](https://github.com/obsproject/obs-studio/commit/c021338ceb75e0e87b0cd6645ba8bf585686dc31)), or [the commit message subject contains the word "Revert" before the rest of the message follows in quotes](https://github.com/obsproject/obs-studio/pull/4641), or [the commit message has no prefix](https://github.com/obsproject/obs-studio/commit/90df8d44dfa71d8439c0b5e6d975dabfd17d93b2), or [the commit message subject has more than 50 characters after the prefix](https://github.com/obsproject/obs-studio/commit/ebbe856d83e5e96870a29c55686a5006d5c71113) because making it shorter makes it non-descriptive and our hard limit is 72 characters total.  All of these would fail to pass this check.

However, this check would, as far as I can tell, pass a commit message like this:
```
: No prefix, but colon still here

.
```

As this is, I don't think I can approve this PR unless all of the cases I mentioned above are handled gracefully.  Either the regex pattern needs to be adjusted, or a PR that fails this check needs to not hold up the rest of the workflow and simply be a warning.  Even in the case of the warning, I would prefer that the regex pattern handle the cases I've outlined above.

File: .github/workflows/pull-request-format.yml
Comment (RytoEX) [inline .github/workflows/pull-request-format.yml:49]: Remove this.

**Review by RytoEX — CHANGES_REQUESTED** [review]
The regex pattern currently used in this PR does not cover all cases of commit messages that we would normally find acceptable and merge.  Some of these are trivial to solve (add another character to the character class), some of these are not as easy.  See my review comment for details.

I concur with @PatTheMav that I don't see the necessity of checking this in CI, though if it can gracefully handle things we'd normally permit without stopping an entire workflow, then I'm not terribly opposed to it either.

File: .github/workflows/pull-request-format.yml
Comment (WizardCM) [inline .github/workflows/pull-request-format.yml:5]: I would argue as this is designed for PRs entirely, this doesn't need to run on `push` at all.

**Comment by gxalpha** [comment]
#5248 looks (while more complicated) way better than this one. Closing.

**Pull request closed without merging**
